components/apimgt/org.wso2.carbon.apimgt.eventing/src/main/java/org/wso2/carbon/apimgt/eventing/EventPublisherFactoryProducer.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.apimgt.eventing;
+
+import java.lang.reflect.InvocationTargetException;
+
+/**
+ * Factory producer for getting an event publisher factory.
+ */
+public class EventPublisherFactoryProducer {
+
+    /**
+     * Private constructor.
+     */
+    private EventPublisherFactoryProducer() {
+
+    }
+
+    /**
+     * Get event publisher factory.
+     *
+     * @param className class name of the event publisher factory
+     * @return event publisher factory
+     */
+    public static EventPublisherFactory getEventPublisherFactory(String className) throws ClassNotFoundException,
+            InstantiationException, IllegalAccessException, NoSuchMethodException, InvocationTargetException {
+        return (EventPublisherFactory) Class.forName(className).getDeclaredConstructor().newInstance();
```

Comment:
Is it correct to create a new EventPublisherFactory instance for each get call?

Reply:
Also do we really need to have a method to pass the class name and initiate a new object? Can you please explain how are we going to use this?
In OSGi this might not be able to use with classes which are not visible to this package.

Reply:
1. new EventPublisherFactory will be created only at the server startup time.
2. the type of the publisher (event hub or azure service bus) will be passed through deployment.toml, EventHubPublisher will be the default one and ASBPublisher will be written in choreo-apimgt-extensions repo.

Reply:
+ classes will be visible to this if we add '*;resolution:=optional' under import packages, right? That's what we do for other packages which might work with external packages?